docs/Supplementary Views.md

```diff
@@ -0,0 +1,76 @@
+### Supplementary Views
+
+Before a supplementary view can be rendered on the screen, it must first be dequeued. This can happen in a number of different ways. When manually conforming to `UICollectionViewDataSource`, this occurs in the `collectionView(_:viewForSupplementaryElementOfKind:at:)` function. If using a `UICollectionViewDiffableDataSource`, this occurs using the `supplementaryViewProvider`.
+
+When implementing these methods, the `UICollectionView` will pass a `String` that represents the kind of element the collection view needs. This element kind if used to differentiate between multiple types of supplementary views, for example you might have both a section header and a section footer for a particular section.
+
+Your collection view layout is what defines the element kind for your supplementary views. Both `UICollectionViewFlowLayout` and `UICollectionLayoutListConfiguration` use built in `String`s for headers (`UICollectionView.elementKindSectionHeader`) and footers (`UICollectionView.elementKindSectionFooter`). If you are building a custom `NSCollectionLayoutSection`, then you must define the element kind when creating either an `NSCollectionLayoutSupplementaryItem` or `NSCollectionLayoutBoundarySupplementaryItem`.
+
+Then, when you register a class or Nib for a supplementary view, you must specify the element kind that you are creating the registration for (whether you are using `UICollectionView.SupplementaryRegistration` or the manual `register` family of methods). You *must* have at least one registration for each element kind that your layout uses.
+
+When the collection view gets to the point where it needs to request a supplementary view for display, it will ask your data source to provide one for the given element kind it needs. For a collection view containing multiple types of supplementary views, each element kind will be requested separately. It is then your responsibility to look at the requested element kind and return a supplementary view for that element kind. It is entirely possible to have multiple `SupplementaryRegistration` or multiple re-use identifiers registered for a single element kind, but you must always ensure to match the requested element kind with the one that is returned.
+
+
+### How UtiliKit Helps
+
+UtiliKit aims to make this process as foolproof as possible. As always, there are 3 steps to correctly implementing the supplementary dequeue process- definition, registration, and dequeue. As an example, we'll walk through a case using a diffable data source and a compositional layout.
+
+1. Definition
```

Comment:
Minor, but should we go ahead and emphasize these steps so that they stand out a bit more when the markdown is rendered? Either `####` or maybe `**bolding**` it?
```suggestion
#### 1. Definition
```

Reply:
Good plan, I agree